lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_state_machine.rb

```diff
@@ -49,6 +49,10 @@ def process_state_step_with_error_handling(f, step = nil)
         $miq_ae_logger.warn("Skipping to state #{@workspace.root['ae_next_state']}")
         @workspace.root['ae_result'] = 'skip' if step == 'on_entry'
       end
+    rescue MiqAeException::StopInstantiation => e
+      error_message = "State=<#{f['name']}> running #{step} raised exception: <#{e.message}>"
+      $miq_ae_logger.error(error_message)
```

Comment:
Suggest moving the text interpolation directly into the `$miq_ae_logger.error` line as the `error_message` is only used once.

The rescue block below is setting a variable because it re-uses the results.